### PR TITLE
cli upgrade + optimizations to generating image versions

### PR DESCRIPTION
**Pull request by anujmaheshwari1**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: vhdbuilder/scripts/automate_generate_version.sh [hunk 1]
```diff
@@ -4,22 +4,23 @@ set -euxo pipefail
 build_ids=$1
 global_image_version="${IMAGE_VERSION:=}"
 for build_id in $build_ids; do
-    for artifact in $(az pipelines runs artifact list --run-id $build_id | jq -r '.[].name'); do    # Retrieve what artifacts were published
+    artifacts=($(az pipelines runs artifact list --run-id $build_id | jq -r '.[].name'))
+    artifacts=(${artifacts[@]//vhd*})
```
Comment (anujmaheshwari1) [inline vhdbuilder/scripts/automate_generate_version.sh:8]: this filters out all the files starting with vhd, so all that is left in the artifacts list is publishing info, which is what we really ever needed.

File: vhdbuilder/scripts/automate_generate_version.sh [hunk 2]
```diff
@@ -4,22 +4,23 @@ set -euxo pipefail
...
+    printf '%s\n' "${artifacts[@]}"
+    for artifact in ${artifacts[@]}; do    # Retrieve what artifacts were published
         # This loop is because of how the Image Version is set for builds. 
         # It uses the UTC time of when the build for a particular SKU ends. 
         # So in the past, it has happened that you trigger a build at say 3/4pm PST, 
         # some SKUs will have todays date some will have tomorrows based on when they are triggered because of UTC conversion
         # TODO(amaheshwari): Change VHD script to use a common var for image version that is plumbed down to all SKUs
-        if [[ $artifact == *"publishing-info"* ]]; then
```
Comment (cameronmeissner) [inline vhdbuilder/scripts/automate_generate_version.sh:13]: might be worth keeping this check for sanity's sake? 

I doubt the artifact names are gonna change anytime soon but would still be nice to have I think, up to you though

Comment (anujmaheshwari1) [reply vhdbuilder/scripts/automate_generate_version.sh:13]: fair enough, since we already got some gains on speed, I will keep this check as is.

Comment (anujmaheshwari1) [reply vhdbuilder/scripts/automate_generate_version.sh]: resolved

**Review by cameronmeissner — APPROVED** [review]

**Pull request closed without merging**